.github/workflows/polyglot_release.yml

```diff
@@ -2,81 +2,49 @@ name: Release Polyglot Piranha
 on:
   workflow_dispatch:
 jobs:
-  build_ubuntu:
-    runs-on: ubuntu-latest
+  release:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, ubuntu-20.04, macos-latest]
+        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
+    runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-python@v4
+    - uses: actions/checkout@v4
+    - uses: actions/setup-python@v5
       with:
-        python-version: 3.8
-    - name: Build wheel with Maturin
+        python-version: ${{ matrix.python-version }}
+    - name: Setup rustup target linux aarch64
+      if: ${{ matrix.os == 'ubuntu-latest' }}
       run: |
-        pip install --upgrade maturin
-        maturin build --release -o dist
-        maturin build --sdist -o dist
-    - name: Pypi Release for ubuntu-latest
+        sudo apt-get update
+        sudo apt-get install gcc-aarch64-linux-gnu
+        rustup target add aarch64-unknown-linux-gnu
+        mkdir -p .cargo
+        touch .cargo/config.toml
+        echo "[target.aarch64-unknown-linux-gnu]" >> .cargo/config.toml
+        echo "linker = \"aarch64-linux-gnu-gcc\"" >> .cargo/config.toml
+    - name: Setup rustup target macOS
+      if: ${{ matrix.os == 'macos-latest' }}
       run: |
-        pip install twine
-        twine upload --skip-existing -u __token__ -p ${{ secrets.PYPI_TOKEN }} dist/*
-  build_ubuntu_20:
-    runs-on: ubuntu-20.04
-    steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-python@v4
-      with:
-        python-version: 3.8
-    - name: Build wheel with Maturin
+        rustup target add x86_64-apple-darwin
+        rustup target add aarch64-apple-darwin
+    - name: pip install --upgrade maturin
       run: |
         pip install --upgrade maturin
+    - name: Build wheel with Maturin linux x86
+      if: ${{ matrix.os == 'ubuntu-20.04' }}
+      run: |
         maturin build --release -o dist
         maturin build --sdist -o dist
```

Comment:
Lets maybe have a separate step for the source distribution? It doesn't seem like it belongs with Linux x86 necessarily.

If we do that, we can combine all matrin steps (other than src) into one.